src/Controls/src/Core/Hosting/AppHostBuilderExtensions.cs

```diff
@@ -9,67 +10,59 @@ namespace Microsoft.Maui.Controls.Hosting
 {
 	public static partial class AppHostBuilderExtensions
 	{
-		static readonly Dictionary<Type, Type> DefaultMauiControlHandlers = new Dictionary<Type, Type>
+		public static IMauiHandlersCollection AddMauiControlsHandlers(this IMauiHandlersCollection handlersCollection)
 		{
-#if __IOS__ || __ANDROID__
-			{ typeof(CollectionView), typeof(CollectionViewHandler) },
-#endif
-
-#if WINDOWS
-			{ typeof(CollectionView), typeof(CollectionViewHandler) },
+#if __IOS__ || __ANDROID__ || WINDOWS
+			handlersCollection.AddHandler<CollectionView, CollectionViewHandler>();
 #endif
-
 #if WINDOWS || __ANDROID__
-			{ typeof(Shell), typeof(ShellHandler) },
+			handlersCollection.AddHandler<Shell, ShellHandler>();
 #endif
-			{ typeof(Application), typeof(ApplicationHandler) },
-			{ typeof(ActivityIndicator), typeof(ActivityIndicatorHandler) },
-			{ typeof(BoxView), typeof(ShapeViewHandler) },
-			{ typeof(Button), typeof(ButtonHandler) },
-			{ typeof(CheckBox), typeof(CheckBoxHandler) },
-			{ typeof(DatePicker), typeof(DatePickerHandler) },
-			{ typeof(Editor), typeof(EditorHandler) },
-			{ typeof(Entry), typeof(EntryHandler) },
-			{ typeof(GraphicsView), typeof(GraphicsViewHandler) },
-			{ typeof(Image), typeof(ImageHandler) },
-			{ typeof(Label), typeof(LabelHandler) },
-			{ typeof(Layout), typeof(LayoutHandler) },
-			{ typeof(Picker), typeof(PickerHandler) },
-			{ typeof(ProgressBar), typeof(ProgressBarHandler) },
-			{ typeof(ScrollView), typeof(ScrollViewHandler) },
-			{ typeof(SearchBar), typeof(SearchBarHandler) },
-			{ typeof(Slider), typeof(SliderHandler) },
-			{ typeof(Stepper), typeof(StepperHandler) },
-			{ typeof(Switch), typeof(SwitchHandler) },
-			{ typeof(TimePicker), typeof(TimePickerHandler) },
-			{ typeof(Page), typeof(PageHandler) },
-			{ typeof(WebView), typeof(WebViewHandler) },
-			{ typeof(Border), typeof(BorderHandler) },
-			{ typeof(IContentView), typeof(ContentViewHandler) },
-			{ typeof(Shapes.Ellipse), typeof(ShapeViewHandler) },
-			{ typeof(Shapes.Line), typeof(ShapeViewHandler) },
-			{ typeof(Shapes.Path), typeof(ShapeViewHandler) },
-			{ typeof(Shapes.Polygon), typeof(ShapeViewHandler) },
-			{ typeof(Shapes.Polyline), typeof(ShapeViewHandler) },
-			{ typeof(Shapes.Rectangle), typeof(ShapeViewHandler) },
-			{ typeof(Shapes.RoundRectangle), typeof(ShapeViewHandler) },
-			{ typeof(Window), typeof(WindowHandler) },
-			{ typeof(ImageButton), typeof(ImageButtonHandler) },
-			{ typeof(IndicatorView), typeof(IndicatorViewHandler) },
+			handlersCollection.AddHandler<Application, ApplicationHandler>();
+			handlersCollection.AddHandler<ActivityIndicator, ActivityIndicatorHandler>();
```

Comment:
I got rid of this `Dicionary<Type,Type>` completely, as the linker can better understand the generic version of `AddHandler<>()`.

Reply:
Lovely!